File: src/main/java/features/reports/pdf/CollectionData.java
package features.reports.pdf;

import core.data.Data;

import java.util.Collection;

/**
 * {@link Data} that represents collection elements
 * @param <T> element type
 */
//TODO: (avlomakin) consider removing 'extends Data' from T
public class CollectionData<T extends Data> extends Data {

    public final Collection<T> collection;

    public CollectionData(Collection<T> collection) {
        this.collection = collection;
    }
}

File: src/main/kotlin/com/mirf/core/data/CollectionData.kt
package com.mirf.core.data

/**
 * [Data] that represents collection elements
 * @param <T> element type
</T> */
//TODO: (avlomakin) consider removing 'extends Data' from T
class CollectionData<T : Data>(val collection: Collection<T>) : MirfData()
